### ISSUE TITLE
Scripts to process ED output for AGB,DBH,PFT abundance,etc

### ISSUE THREAD

**Issue by fabeit** [EDmodel/ED2#172]
I have been using @mpaiao scripts in the ED/template/ folder to process ED output, however because these are old and variables in the h5 files have changed I had to comment and made quite a few changes, with some issues that I have not been able to address.

I am mostly interested in analyzing dbh class size distribution, pft composition, agb time series, height time series, etc.
Does anybody have any script that does these kind of things, and willing to share before I reinvent the wheel? @mpaiao scripts are very extensive but I am wondering if other people have more recent scripts that focus on the aspects I have mentioned.

**Comment by DanielNScott** [comment]
Yeah my output postproccessing stuff will either do that or would with minor modification. I automatically look at DBH growth by pft when I do simulations. You can get my scripts from my ed_tools repo if you like, and I'd be happy to show you how they work, figure out if/how they can be modified, or whatever.

Regardless of what you decide, I think this hits on an important issue of script sharing which it might be valuable to have a larger conversation about (during the code meeting maybe?).

I think it would be good to have 'canonical' ED input and output processing and graphing utilities, which at this point I believe @mpaiao (via his R scripts), @mdietze's group (via pecan), @rgknox (via EDTS), @ke-zhang, and myself have all 'reinvented' in some form or another... I've brought this up with Paul and he's been uncertain about it. So I guess let me take the opportunity to ask you guys, is trying to coordinate/consolidate this stuff

1) a waste because people should build their own tools and thereby gain facility w/ ED
2) a waste because e.g. Marcos' R code is so awesome, or 
3) something that should happen?

Thoughts? Or defer the conversation?

**Comment by fabeit** [comment]
@DanielNScott thanks for the feedback, I have looked at your scripts but I don't think I will be able to use them for now, I never worked with matlab and I don't have the license. But I completely agree we could have a collection of scripts (R, matlab, etc) to process ED output. Marcos' code is really great it just needed a bit of updating, now I got it to work and I can commit them but before I think we should have this discussion.

**Comment by mdietze** [comment]
I'd definitely like to have a conversation about this. I firmly reject #1 "people should build their own tools". My group is mostly using PEcAn, but since building PEcAn we haven't invested in any good visualization tools for cohort-level information, focusing more on overall pools and fluxes. However, the number of models within PEcAn that have cohort-level info is growing rapidly (CLM-ED, LPJ-GUESS, MAESPA, LANDIS, LINKAGES, etc) and we'd really like to generalize some of what the ED community's done to support visualization, benchmarking, and assimilation of cohort-level info.
